### PR TITLE
Document the 'reviewer' role

### PR DESCRIPTION
**Pull request by negz**
### Description of your changes

Pertains to https://github.com/crossplane/crossplane/issues/3046

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

[contribution process]: https://git.io/fj2m9

I've done a brief proof-read.

### CODE REVIEW

File: GOVERNANCE.md
Comment (hasheddan) [inline GOVERNANCE.md:352]: Might be worth linking to the conflict resolution section, I was expecting it would come in the subsequent section when reading here.

**Review by hasheddan — APPROVED** [review]

**Review by muvaf — APPROVED** [review]
LGTM!

**Comment by negz** [comment]
@bassam @jbw976 I need an approval from one of you on this one since you're now code owners for all governance and charter related documents.

**Review by bassam — APPROVED** [review]

**Comment by negz** [comment]
Thanks!

**Pull request merged**
